scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/package.scala

```diff
@@ -42,27 +40,30 @@ package object avro extends Syntax {
   /** Alias for `me.lyh.parquet.avro.Predicate`. */
   val Predicate = me.lyh.parquet.avro.Predicate
 
-  class ParquetAvroSink[T](
-    schema: Schema,
-    val compression: CompressionCodecName,
-    val conf: SerializableConfiguration
-  ) extends FileIO.Sink[T] {
-    private val schemaString = schema.toString
-    private var writer: ParquetWriter[T] = _
-    override def open(channel: WritableByteChannel): Unit = {
-      val schema = new Schema.Parser().parse(schemaString)
-      // https://github.com/apache/parquet-mr/tree/master/parquet-hadoop#class-parquetoutputformat
-      val rowGroupSize =
-        conf.get.getInt(ParquetOutputFormat.BLOCK_SIZE, ParquetWriter.DEFAULT_BLOCK_SIZE)
-      writer = AvroParquetWriter
-        .builder[T](new ParquetOutputFile(channel))
-        .withSchema(schema)
-        .withCompressionCodec(compression)
-        .withConf(conf.get)
-        .withRowGroupSize(rowGroupSize)
-        .build
+  def classIsSpecificRecord[T: ClassTag]: Boolean = {
+    val avroClass: Class[T] = ScioUtil.classOf[T]
+    classOf[SpecificRecord] isAssignableFrom avroClass
+  }
+
+  implicit class ConfigurationSyntax(conf: Configuration) {
+
+    def setReadSchemas[A, T](params: ParquetAvroIO.ReadParam[A, T]): Unit = {
```

Comment:
What do you think of broadening the scope of this method and moving into a class method of `ParquetAvroIO.ReadParam`? Like:

```scala
 final case class ReadParam[A: ClassTag, T: ClassTag] private (
    projectionFn: A => T,
    projection: Schema = ReadParam.DefaultProjection,
    predicate: FilterPredicate = ReadParam.DefaultPredicate,
    conf: Configuration = ReadParam.DefaultConfiguration,
    suffix: String = null
  ) {
   val avroClass: Class[A] = ScioUtil.classOf[A]
   val isSpecific = classOf[SpecificRecord] ScioUtil.classOf[T] 
   
   private[avro] def getReadConfiguration(): Configuration = {
       val c = ParquetConfiguration.ofNullable(conf)
       // Set schema/projection
       val readSchema: Schema = if (isSpecific) ReflectData.get().getSchema(avroClass) else params.projection

       AvroReadSupport.setAvroReadSchema(readSchema)
       if (params.projection != null) { /* set projection... */ }
       if (params.predicate != null) { /* set predicate */ }
   
       // Set data supplier
       if (!isSpecific && conf.get(AvroReadSupport.AVRO_DATA_SUPPLIER) == null) {
         ....
      }
  }
```

That way it's easy to access from both `read` and `tap` methods 👍

Reply:
This is great! I missed it, the great place for common functions on input conf!